py-powsybl/src/main/java/com/powsybl/powsybl/PythonCodeGenerator.java

```diff
@@ -0,0 +1,267 @@
+            return returnElementName.substring(0, returnElementName.length() - 2);
+        }
+        if (removeS.equals("Shunt")) {
+            return "ShuntCompensator";
+        }
+        return removeS;
+    }
+
+    static boolean skipByReturnType(Method m) {
+        String returnTypeSimpleName = m.getReturnType().getSimpleName();
+        if (m.getReturnType().toString().contains("$")) {
+            System.out.println(m + " TODO nested class");
+            return true;
+        }
+        if (returnTypeSimpleName.endsWith("TapChanger")) {
+            System.out.println(m + " TODO tapchanger");
+            return true;
+        }
+        if (returnTypeSimpleName.endsWith("StateManager")) {
+            System.out.println(m + " TODO StateManager");
+            return true;
+        }
+        return false;
+    }
+
+    static boolean skip(Method m, Class clazz) {
+        String methodName = m.getName();
+        List<Annotation> annotations = Arrays.asList(m.getDeclaredAnnotations());
+        for (Annotation anno : annotations) {
+            if (anno.annotationType().getSimpleName().equals("Deprecated")) {
+                System.out.println(m + " skipped(Deprecated)");
+                return true;
+            }
+        }
+        if (methodName.startsWith("new")) {
+            System.out.println(m + " skipped(new)");
+            return true;
+        }
+        if (methodName.endsWith("Stream")) {
+            System.out.println(m + " skipped(Stream)");
+            return true;
+        }
+        if (methodName.startsWith("set") && !clazz.getSimpleName().equals("Switch")) {
+            System.out.println(m + " skipped(set)");
+            return true;
+        }
+        List<Class> parameterClass = Arrays.asList(m.getParameterTypes());
+        for (Class c : parameterClass) {
+            if (SKIP_PARA_TYPE.contains(c.getSimpleName())) {
+                System.out.println(m + " skipped(parameters type)");
+                return true;
+            }
+        }
+        for (String mn : SKIP_METHODS_NAME) {
+            if (methodName.startsWith(mn)) {
+                System.out.println(m + " skipped(mapbe to implement later)");
+                return true;
+            }
+        }
+
+        List<Class> superInterfaces = Arrays.asList(clazz.getInterfaces());
+        if (!superInterfaces.isEmpty()) {
+            List<Method> methodsInSuperClass = Arrays.asList(superInterfaces.get(0).getMethods());
+            if (methodsInSuperClass.contains(m)) {
+                System.out.println(m + " skipped(super method)");
+                return true;
+            }
+        }
+        return false;
+    }
+
+    static String to_python_style(String javaMethodName) {
+        if (javaMethodName.equals("getbPerSection")) {
+            return "get_b_per_section";
+        }
+        String[] r = javaMethodName.split("(?=\\p{Upper}|\\d)");
```

Comment:
It works fine so it's OK, but maybe it would have been even easier with guava [CaseFormat](https://google.github.io/guava/releases/20.0/api/docs/com/google/common/base/CaseFormat.html) !

Reply:
CaseFormat returns get_terminal1 not get_terminal_1